server-fn.c

```diff
@@ -438,15 +443,31 @@ server_destroy_session(struct session *s)
 		s_new = session_previous_session(s);
 	else if (detach_on_destroy == 4)
 		s_new = session_next_session(s);
+
 	if (s_new == s)
 		s_new = NULL;
```

Comment:
Is this check a legacy remnant or is it kept as an extra security against potential future issues?
As far as I can tell none of the replacement-session-seeking functions could return the original `s`ession.

Reply:
It is probably left over from old code.